src/gluonts/model/deepar/_estimator.py

```diff
@@ -127,118 +137,63 @@ class DeepAREstimator(GluonEstimator):
-            [c > 0 for c in cardinality]
-        ), "Elements of `cardinality` should be > 0"
-        assert embedding_dimension is None or all(
-            [e > 0 for e in embedding_dimension]
-        ), "Elements of `embedding_dimension` should be > 0"
-        assert (
-            num_parallel_samples > 0
-        ), "The value of `num_parallel_samples` should be > 0"
-        assert alpha >= 0, "The value of `alpha` should be >= 0"
-        assert beta >= 0, "The value of `beta` should be >= 0"
+    freq: str
+    prediction_length: ty.PositiveInt
+    alpha: ty.PositiveFloat0 = 0.0
+    beta: ty.PositiveFloat0 = 0.0
+    cardinality: List[ty.PositiveInt] = [1]
+    cell_type: str = "lstm"
+    context_length: Optional[ty.PositiveInt] = None
+    distr_output: DistributionOutput = StudentTOutput()
+    dropout_rate: ty.PositiveFloat0 = 0.1
+    dropoutcell_type: DropoutcellType = DropoutcellType.ZoneoutCell
+    dtype: DType = np.float32
+    embedding_dimension: Optional[List[ty.PositiveInt]] = None
+    imputation_method: Optional[MissingValueImputation] = None
+    lags_seq: Optional[List[int]] = None
+    num_cells: ty.PositiveInt = 40
+    num_layers: ty.PositiveInt = 2
+    num_parallel_samples: ty.PositiveInt = 100
+    scaling: bool = True
+    time_features: Optional[List[TimeFeature]] = None
+    train_sampler: InstanceSampler = ExpectedNumInstanceSampler(1.0)
+    trainer: Trainer = Trainer()
+    use_feat_dynamic_real: bool = False
+    use_feat_static_cat: bool = False
+    use_feat_static_real: bool = False
 
-        self.freq = freq
-        self.context_length = (
-            context_length if context_length is not None else prediction_length
-        )
-        self.prediction_length = prediction_length
-        self.distr_output = distr_output
-        self.distr_output.dtype = dtype
-        self.num_layers = num_layers
-        self.num_cells = num_cells
-        self.cell_type = cell_type
-        self.dropoutcell_type = dropoutcell_type
-        self.dropout_rate = dropout_rate
-        self.use_feat_dynamic_real = use_feat_dynamic_real
-        self.use_feat_static_cat = use_feat_static_cat
-        self.use_feat_static_real = use_feat_static_real
-        self.cardinality = (
-            cardinality if cardinality and use_feat_static_cat else [1]
-        )
-        self.embedding_dimension = (
-            embedding_dimension
-            if embedding_dimension is not None
-            else [min(50, (cat + 1) // 2) for cat in self.cardinality]
-        )
-        self.scaling = scaling
-        self.lags_seq = (
-            lags_seq
-            if lags_seq is not None
-            else get_lags_for_frequency(freq_str=freq)
-        )
-        self.time_features = (
-            time_features
-            if time_features is not None
-            else time_features_from_frequency_str(self.freq)
-        )
+    def __init__(self, **fields):
+        BaseModel.__init__(self, **fields)
```

Comment:
Can one pass anything here to the constructor? Or just the fields declared in the class?

Reply:
You could pass anything there, but only the fields which are declared are handled.